mlflow/projects/_project_spec.py

```diff
@@ -154,7 +155,12 @@ def compute_parameters(self, user_parameters, storage_dir):
 
         for key, param_obj in self.parameters.items():
             value = user_parameters[key] if key in user_parameters else self.parameters[key].default
-            final_params[key] = param_obj.compute_value(value, storage_dir)
+            target_sub_dir = 'param_{}'.format(list(self.parameters.keys()).index(key))
+            download_dir = None
+            if storage_dir:
+                download_dir = os.path.join(storage_dir, target_sub_dir)
+                os.mkdir(download_dir)
```

Comment:
@trangevi By moving the directory creation call into `compute_parameters`, we're now creating a directory for each parameter, even if the parameter isn't of type `path` (e.g., we have directories for `string` and `float` params for which we won't be downloading any content). Can we move the `download_dir` creation logic back into `_compute_path_value()`? We can add another `param_index` parameter to the method that can be used to determine the working directory name.